### DISCUSSION TITLE
How to use Lipgloss to render tree of objects?

### DISCUSSION THREAD

**Discussion by quenbyako** [charmbracelet/lipgloss#171] [Q&A]
Hi, i worked a lot with flutter framework, which allows you to build tree of elements to render them in area constraints (like min/max width/height, preferences in position, align, etc etc etc). I trying to do same stuff with Lipgloss, to make rows, columns and tables in my app (as example, i trying to find some references in flutter and html structures)

Is it even possible with current version of Lipgloss to insert into style some child components with specific position, align etc? If yes, do you have any tips, how to implement it?

For now, as an alternative, i trying to create additional package (and planning also to create pull request or separated open repository), which will handle all stuff related to tree structure rendering, but i hope that there is some builtin way to do it without pain (which will be super cool!)

**Comment by MichaelMure** [comment]
I suppose you are more looking for CSS grids, but maybe flexbox would help? https://github.com/charmbracelet/lipgloss/issues/166

That code is not fully done but it's already working quite well.

**Reply by quenbyako** [reply]
It looks similar to what i want (and thanks for opening an issue about that!), but honestly i expected to find some set of elements/widgets or similar thing, and use them in my cli interface. 

Maybe a single way to have is to write by my hands 😥
